task_processing/interfaces/task_executor.py

```diff
@@ -24,14 +21,14 @@ class TaskExecutor(object):
     """
 
     @abc.abstractmethod
-    def run(self, task_config):
-        """Run the supplied task
+    def run(self, task_config, task_id):
```

Comment:
we need to pass it separately to make sure task_config is not mutated on the way and it's a vector to reduce the assumptions every executor has to make and share about format of task id, how it's mutated in the stack, how to break it up into components, etc. by having task_id a vector all other executors need to know is that they can grow/shrink it on the tail, that's it.